corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -32,31 +36,65 @@ def start_reindex(self, cname):
             raise IndexNotMultiplexedException("""Index not multiplexed!
             Sync can only be run on multiplexed indices""")
 
-        self.source, self.dest = self._get_source_destination_indexes(adapter)
-
-        task_id = es_manager.reindex(self.source, self.dest)
-
+        source_index, destination_index = self._get_source_destination_indexes(adapter)
+
+        logger.info(f"Preparing index {destination_index} for reindex")
+        self._prepare_index_for_reindex(destination_index)
+
+        logger.info("Starting ReIndex process")
+        task_info = es_manager.reindex(source_index, destination_index)
+        logger.info(f"Copying docs from index {source_index} to index {destination_index}")
+        task_id = task_info.split(':')[1]
+        print("\n\n\n")
+        logger.info("-----------------IMPORTANT-----------------")
+        logger.info(f"TASK ID - {task_id}")
+        logger.info("-------------------------------------------")
+        logger.info("Save this Task Id, You will need it later for verifying your reindex process")
+        print("\n\n\n")
         # This would display progress untill reindex process is completed
-        check_task_progress(task_id)
+        check_task_progress(task_info)
 
+        print("\n\n")
         self.perform_cleanup(adapter)
 
+        logger.info("Preparing Index for normal use")
+        self._prepare_index_for_normal_usage(adapter.secondary)
```

Comment:
Is it safe to do these things before verifying that the reindex is complete?

I was expecting a prompt for user verification of reindex completion prior to proceeding with these steps.

Reply:
In terms of safety it should be safe, as it will only increase number of replicas and would reduce refresh interval of the index.
If the reindex will fail, it should fail hard and will not reach here. But there can be a case when it can reach this point without actually finishing reindex and  that is if the task is cancelled by someone(or maybe by ES itself).
However in that case too the only downside is that that there would be ES will have to reset the replicas to 0 and i.e remove the data in configured replicas and then start the reindex which I assume should be safe too.

Reply:
Fair enough. My only concern is that operators may not be expecting the index to be automatically reconfigured for standard operation if they cancel the reindex. I don't think it's worth blocking this PR on it though.